controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -198,7 +220,7 @@ func (r *DataScienceClusterReconciler) reconcileSubComponent(instance *dsc.DataS
 	})
 	if err != nil {
 		instance = r.reportError(err, instance, "failed to update DataScienceCluster conditions before reconciling "+componentName)
-		// try to continue with reconciliation, as further updates can fix the status
+		return instance, err
```

Comment:
I was back and forth on this one, between returning or continuing. In the end, I feel that since only the status update failed, continuing feels like a better option as the status will be updated again right after the component is reconciled.